src/Mod/Sketcher/Gui/EditModeConstraintCoinManager.cpp

```diff
@@ -669,10 +669,15 @@ void EditModeConstraintCoinManager::processConstraints(const GeoListFacade & geo
 
                             } else if (geo->getTypeId() == Part::GeomCircle::getClassTypeId()) { // circle to circle distance
                                 const Part::Geometry *geo1 = geolistfacade.getGeometryFromGeoId(Constr->First);
-                                auto circleSeg1 = static_cast<const Part::GeomCircle*>(geo1);
+                                const Part::GeomCircle *circleSeg1;
+                                if (geo1->getTypeId() == Part::GeomCircle::getClassTypeId())
+                                    circleSeg1 = static_cast<const Part::GeomCircle*>(geo1);
+                                else
+                                    break;
+
                                 auto circleSeg2 = static_cast<const Part::GeomCircle*>(geo);
 
-                                GetCirclesMinimalDistance(circleSeg1, circleSeg2, &pnt1, &pnt2);
+                                GetCirclesMinimalDistance(circleSeg1, circleSeg2, pnt1, pnt2);

```

Comment:
I do not see an advantage of this over a single if statement as I indicated in the other comment. Because circleSeg1 is already of type GeomCircle, it cannot be reused in the else branch for an GeomArcOfCircle.

I think it is less readable like this, as if is not immediately clear when the circleSeg2 and the function are called.

If you really want to leave arc ready code in this PR as an exercise, you could template the function and try to write compiling code for Circles and ArcsOfCircle here. I think you will soon realise the limitations.

Yet an alternative, in Part::GeomCurve (Geometry.h and Geometry.cpp in Part Mod), you write a member function:
```void closestParameters(const GeomCurve * other, double&u1, double&u2)```

Then using OCCT you calculate the parameters in the curve and in the other curve (u1 and u2) corresponding to the closest points.

This u1 and u2, you pass them via Part::pointAtParameter (of the corresponding geomcurve). Now you have support for all geomcurves, and you can just write here:

if(geo is GeomCurve and geo1 is GeomCurve) 
call those two functions to get the points.

If you want to do this, take a look to ```GeomCurve::intersect``` which uses OCCT ```GeomAPI_ExtremaCurveCurve ```. However, this will also come with additional caveats, as extrema (read the docs) sometimes does not provide the closest distance between curves, and for this specific use case, you would need distance between basis curves (non-trimmed curves) anyway...

So, probably now you are in a best position to select:
(a) I will keep this PR minimal and do a minimal readable call to the function I have.
(b) I want to learn function templates, I was waiting for this opportunity for a long time.
(c) I am getting bored and this OCCT world looks amasing...

In any case, I would still choose (a) and do another PR targeting (b) or (c), if you feel like  :)

Reply:
I follow your wise advice.